pilot/pkg/model/validation.go

```diff
@@ -1813,6 +1815,25 @@ func ValidateVirtualService(_, _ string, msg proto.Message) (errs error) {
 					errs = appendErrors(errs, fmt.Errorf("duplicate hosts in virtual service: %s & %s", hostI, hostJ))
 				}
 			}
+
+			if !strings.Contains(virtualService.Hosts[i], ".") {
+				svcShortNames = append(svcShortNames, virtualService.Hosts[i])
+			} else if strings.Contains(virtualService.Hosts[i], "."+namespace+"."+"svc") {
+				svcFQDNs = append(svcFQDNs, virtualService.Hosts[i])
+			}
+		}
+
+		// Check for multi services short name.
+		if len(svcShortNames) > 1 {
+			errs = appendErrors(errs, fmt.Errorf("multi service names specified in virtual service hosts: %v", virtualService.Hosts))
```

Comment:
what does this check enforce?

Reply:
Donot allow setting two diff k8s services, like svcA and svcB, which is meaningless.